### DISCUSSION TITLE
Figma-komponenter vs Code

### DISCUSSION THREAD

**Discussion by Slaine2** [fremtind/jokul#2281] [Annet]
**Figma-komponenter viser i jøkul med iFrame**
Jeg synes det kan være spennende å vise figmas interaktive komponent i Jøkul-biblioteket. Dette er fordi det alltid er uoverensstemmelser mellom de utviklede komponentene og de vi bruker som designere. Leiv-Frederic og jeg har testet dette ved hjelp av en iFrame. Men det kan være enda bedre alternativer som bedre støtter respons.

Hvilke steg i [prosessen](https://fremtind.github.io/jokul/komigang/prosessen) er utført?
- [x] Undersøke
- [ ] Rapportere
- [ ] Avklare
- [ ] Spesifisere krav
- [ ] Utvikle
- [ ] Ferdig!

**Comment by wkillerud** [comment]
Dette kunne vært spennende å få til 😄 Jeg vet ikke hvordan det er med tilganger til Jøkul-biblioteket i Figma, men det bør kunne la seg løse.

Figma-noob her, men [fra hva jeg kan se kan vi bare embede Pages](https://help.figma.com/hc/en-us/articles/360040531773#Links)? Så mulig vi må strukturere om så vi kan embedde riktig innhold per side, eller lage en egen "Portal-fil" som bare brukes til dette. Figma har en [guide til iframe embeds her](https://www.figma.com/developers/embed).

Et alternativ i Gatsby er å bruke Figma som en _source_ [ved hjelp av `gatsby-source-figma`](https://github.com/fabe/gatsby-source-figma). Da ser det ut som Gatsby bruker [Figma sitt API til å lage bilder av hvordan Figma ser ut der og da](https://www.figma.com/blog/introducing-figmas-platform/#how-it-works), og kan hente ut enkeltelementer fra en Page. En ulempe med dette er at [det skjer "build time"](https://www.gatsbyjs.com/docs/conceptual/data-fetching/#source-data-to-be-queried-at-build-time), som betyr at det vi viser fra Figma ikke oppdateres uten en endring i Jøkul-koden. Det kan vi kanskje jobbe oss rundt ved å sette opp [webhooks i Figma (fremdeles i beta)](https://www.figma.com/developers/api#webhooks_v2) som starter et bygg av portalen ved endring i Figma-fila for Jøkul.

**Reply by onordbo** [reply]
Jeg tror det er kult å undersøke live embed, jeg undrer på om vi har et behov for å strekke oss så langt som at vi konstant bygger nye versjoner i sanntid. Oppdaterer Figma-biblioteket seg så ofte at brukere som er inne i portalen har behov for det? 😂

**Comment by onordbo** [comment]
Jeg synes dette er en kul ide, jeg er nysgjerrig på å utforske bruksområdet og hva slags verdi vi ønsker å tilføre brukeropplevelsen ved å integrere live-figma filer i portalen. Det finnes jo en del [dokumentasjon](https://www.figma.com/developers/api) for å integrere mot Figma, og for noen år siden så lanserte Figma en versjon av Web API'et sitt med en slags token-sak som lar oss programmatisk trekke ut innhold fra en figma fil.

Spørsmålet mitt er hva som er mest hensiktsmessig å bruke her? Er det et underliggende behov som driver dette behovet? Hva slags problem løser vi ved å dele live Figma i Jøkul portalen? Er det for å skape en bedre opplevelse for designerene i portalen som single-source of truth? Hva er forskjellen mellom å vise en interaktiv Figma komponent ved siden av en interaktiv implementert komponent i portalen?
